### DISCUSSION TITLE
Reading and writing specific data from a queue

### DISCUSSION THREAD

**Discussion by ghost** [pika/pika#1442] [Q&A]
Hello,

I'm very new to coding and currently working on a project to automatically create and populate queues and shovels, or precisely the other way around, to read data from these two.

Initially, all IPs with RabbitMQ installed are being captured, and the queues contained on them are being read.

Queue name = Shovel name

However, I am struggling to read the following data from or into them:

From the queue: Type, Features, and from the queue's bindings, "from" and "Routingkey."

And the following from the shovel: Protocol, URI, Endpoint (for both Source and Shovel).

This is my current approach:

def read_rabbitmq_data(ip_address):
    credentials = pika.PlainCredentials('guest', 'guest') 
    parameters = pika.ConnectionParameters(ip_address, '/', credentials)
    
    connection = pika.BlockingConnection(parameters)
    channel = connection.channel()

    # Informationen ueber den Clusterstatus abrufen
    cluster_status = channel.cluster_status()
    queues = channel.queue_declare(queue='', passive=True)
    queue_list = []
    for queue_info in queues:
        row_data = []
        row_data[0] = queue_info[0] #name
        row_data[1] = queue_info[1] #Type
        row_data[2] = queue_info[14] #Features
        
        # Bindings abrufen
        binding = channel.queue_bindings(queue=row_data[0])
        row_data[3] = binding['source'] #from
        row_data[4] = binding['routing_key'] #routingkey

    # Shovel-Daten auslesen
        shovels = channel.shovel_list(vhost='/')
        row_data[5] = shovels['name']
        row_data[6] = shovels['Protocol']['source']
        row_data[7] = shovels['Uri']['source']
        row_data[8] = shovels['endpoint']['source']
        row_data[9] = shovels['Protocol']['Destination']
        row_data[10] = shovels['Uri']['Destination']
        row_data[11] = shovels['endpoint']['Destination']

        row_data.insert(queue_list)
    connection.close()

Thank you very much for the help.

**Comment by lukebakken** [comment]
I have this bookmarked to investigate. I have been very busy supporting RabbitMQ customers who pay for support.

**Comment by lukebakken** [comment]
What would help me out is if you could provide a git repository with a **complete** program that I can run, instead of the non-running snippet you have provided.
